types/wordpress__block-editor/index.d.ts

```diff
@@ -12,7 +12,11 @@ declare module "@wordpress/data" {
     function select(key: "core/block-editor"): typeof import("./store/selectors");
 }
 
-export const store: any;
+export interface BlockEditorStoreDescriptor extends StoreDescriptor {
+    name: "core/block-editor";
+}
+
+export const store: BlockEditorStoreDescriptor;
```

Comment:
will this lead to type errors for people trying to access other properties than are exposed in `name` or in `StoreDescriptor`? or, is `StoreDescriptor` exhaustive?

Reply:
Below is what `StoreDescriptor` type is defined as in `@wordpress/data` package.

![image](https://github.com/DefinitelyTyped/DefinitelyTyped/assets/2236554/7735aadf-1b04-4cf0-9524-35c756133465)

https://github.com/WordPress/gutenberg/blob/0bee15148fe4330c20cf372cb46a33693e45cb5f/packages/data/src/types.ts#L21-L31

What other properties people are trying (or will try to) access from store? Can you show me some examples which I can refer?

I only assumed that store object can only be used when you need to pass the store parameter to `select(...)` function or `useSelect(...)` hook from `@wordpress/data` package. In that use case, I've already tested it in my local project and it seems to be working well.

Reply:
> Can you show me some examples which I can refer?

I was only intending to ask if there _are_ other properties, or if `StoreDescriptor` itself is supposed to be comprehensive. if that's the case then this seems fine.

My only concern is creating new types which artificially restrict valid code; everything else that helps out is good.

Reply:
Absolutely. Agreed. That's why I extended it from existing `StoreDescriptor` type rather than creating new one.

If anything changes in `StoreDescriptor`, type for `blockEditorStore` would inherently receive it.

And long term, once `@wordpress/block-editor` package starts publishing its own types, we won't need this anymore.